server/processor.go

```diff
@@ -59,29 +55,23 @@ func newPostProcessor(
 	return &PostProcessor{
 		botID:            botID,
 		moderator:        moderator,
-		stopChan:         make(chan bool, 1),
 		thresholdValue:   thresholdValue,
 		excludedUsers:    excludedUsers,
 		excludedChannels: excludedChannels,
+		posts:            make(chan *model.Post, maxProcessingQueueSize),
 	}, nil
 }
 
 func (p *PostProcessor) start(api plugin.API) {
 	go func() {
 		for {
-			select {
-			case <-p.stopChan:
+			post, ok := <-p.posts
```

Comment:
I think you'll find this easier to have a separate channel for indicating "done".  It is common to have a channel called "done" for this.  The channel's only purpose is to be closed when processing should stop.  This will also avoid the send on close channel panic in the enqueue method.

Reply:
> we could still randomly select the case that sends another post without logging it.

@davidkrauser  I'm not seeing how an extra post getting processed after `stop()` is called would not get logged.

Reply:
We talked about this offline, and will stick with this solution with a small tweak: we want to make sure panics aren't swallowed completely. So I've added more detail to the log statements.